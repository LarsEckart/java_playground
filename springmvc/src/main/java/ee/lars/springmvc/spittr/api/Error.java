package ee.lars.springmvc.spittr.api;

public class Error {

    private final int code;
    private final String message;

    public Error(int code, String message) {
        this.code = code;
        this.message = message;
    }

    public int getCode() {
        return this.code;
    }

    public String getMessage() {
        return this.message;
    }
}