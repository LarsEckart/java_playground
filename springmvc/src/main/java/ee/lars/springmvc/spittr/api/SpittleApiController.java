package ee.lars.springmvc.spittr.api;

import ee.lars.springmvc.spittr.Spittle;
import ee.lars.springmvc.spittr.data.SpittleRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/spittles")
public class SpittleApiController {

    private static final String MAX_LONG_AS_STRING = "9223372036854775807";

    private final SpittleRepository spittleRepository;

    @Autowired
    public SpittleApiController(SpittleRepository spittleRepository) {
        this.spittleRepository = spittleRepository;
    }

    @RequestMapping(method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
    public List<Spittle> spittles(
            @RequestParam(value = "max", defaultValue = MAX_LONG_AS_STRING) long max,
            @RequestParam(value = "count", defaultValue = "20") int count)
    {
        return this.spittleRepository.findSpittles(max, count);
    }

    @RequestMapping(method = RequestMethod.POST, consumes = "application/json")
    public Spittle saveSpittle(@RequestBody Spittle spittle) {
        return spittleRepository.save(spittle);
    }

    @RequestMapping(value = "/{id}", method = RequestMethod.GET)
    public ResponseEntity<?> spittleById(@PathVariable long id) {
        Spittle spittle = this.spittleRepository.findOne(id);
        if (spittle == null) {
            Error error = new Error(4, "Spittle [" + id + "] not found");
            return new ResponseEntity<Error>(error, HttpStatus.NOT_FOUND);
        }
        return new ResponseEntity<Spittle>(spittle, HttpStatus.OK);
    }
}
