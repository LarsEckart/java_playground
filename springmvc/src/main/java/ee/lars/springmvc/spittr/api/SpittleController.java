package ee.lars.springmvc.spittr.api;

import ee.lars.springmvc.spittr.Spittle;
import ee.lars.springmvc.spittr.data.SpittleRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.List;

@Controller
@RequestMapping("/spittles")
public class SpittleController {

    private static final String MAX_LONG_AS_STRING = "9223372036854775807";

    private final SpittleRepository spittleRepository;

    @Autowired
    public SpittleController(SpittleRepository spittleRepository) {
        this.spittleRepository = spittleRepository;
    }

    @RequestMapping(method = RequestMethod.GET)
    public List<Spittle> spittles(
            @RequestParam(value = "max", defaultValue = MAX_LONG_AS_STRING) long max,
            @RequestParam(value = "count", defaultValue = "20") int count)
    {
        return this.spittleRepository.findSpittles(max, count);
    }
}
