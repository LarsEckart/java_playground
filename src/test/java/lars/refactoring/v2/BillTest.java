package lars.refactoring.v2;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class BillTest {

    @Test
    void foo() {
        /*
        assertThat(statement).isEqualTo("Statement for BigCo\n"
                + "  Hamlet: $650.00 (55 seats)\n"
                + "  As You Like It: $580.00 (35 seats)\n"
                + "  Othello: $500.00 (40 seats)\n"
                + "Amount owed is $1,730.00\n"
                + "You earned 47 credits");
         */
    }
}
