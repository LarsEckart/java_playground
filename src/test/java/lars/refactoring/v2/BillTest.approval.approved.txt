[hamlet, 5] => Statement for BigCo
  Hamlet: $400.00 (5 seats)
Amount owed is $400.00
You earned 0 credits 
[hamlet, 10] => Statement for BigCo
  Hamlet: $400.00 (10 seats)
Amount owed is $400.00
You earned 0 credits 
[hamlet, 15] => Statement for BigCo
  Hamlet: $400.00 (15 seats)
Amount owed is $400.00
You earned 0 credits 
[hamlet, 20] => Statement for BigCo
  Hamlet: $400.00 (20 seats)
Amount owed is $400.00
You earned 0 credits 
[hamlet, 25] => Statement for BigCo
  Hamlet: $400.00 (25 seats)
Amount owed is $400.00
You earned 0 credits 
[hamlet, 30] => Statement for BigCo
  Hamlet: $400.00 (30 seats)
Amount owed is $400.00
You earned 0 credits 
[hamlet, 35] => Statement for BigCo
  Hamlet: $450.00 (35 seats)
Amount owed is $450.00
You earned 5 credits 
[hamlet, 40] => Statement for BigCo
  Hamlet: $500.00 (40 seats)
Amount owed is $500.00
You earned 10 credits 
[hamlet, 45] => Statement for BigCo
  Hamlet: $550.00 (45 seats)
Amount owed is $550.00
You earned 15 credits 
[as-like, 5] => Statement for BigCo
  As You Like It: $315.00 (5 seats)
Amount owed is $315.00
You earned 1 credits 
[as-like, 10] => Statement for BigCo
  As You Like It: $330.00 (10 seats)
Amount owed is $330.00
You earned 2 credits 
[as-like, 15] => Statement for BigCo
  As You Like It: $345.00 (15 seats)
Amount owed is $345.00
You earned 3 credits 
[as-like, 20] => Statement for BigCo
  As You Like It: $360.00 (20 seats)
Amount owed is $360.00
You earned 4 credits 
[as-like, 25] => Statement for BigCo
  As You Like It: $500.00 (25 seats)
Amount owed is $500.00
You earned 5 credits 
[as-like, 30] => Statement for BigCo
  As You Like It: $540.00 (30 seats)
Amount owed is $540.00
You earned 6 credits 
[as-like, 35] => Statement for BigCo
  As You Like It: $580.00 (35 seats)
Amount owed is $580.00
You earned 12 credits 
[as-like, 40] => Statement for BigCo
  As You Like It: $620.00 (40 seats)
Amount owed is $620.00
You earned 18 credits 
[as-like, 45] => Statement for BigCo
  As You Like It: $660.00 (45 seats)
Amount owed is $660.00
You earned 24 credits 
[othello, 5] => Statement for BigCo
  Othello: $400.00 (5 seats)
Amount owed is $400.00
You earned 0 credits 
[othello, 10] => Statement for BigCo
  Othello: $400.00 (10 seats)
Amount owed is $400.00
You earned 0 credits 
[othello, 15] => Statement for BigCo
  Othello: $400.00 (15 seats)
Amount owed is $400.00
You earned 0 credits 
[othello, 20] => Statement for BigCo
  Othello: $400.00 (20 seats)
Amount owed is $400.00
You earned 0 credits 
[othello, 25] => Statement for BigCo
  Othello: $400.00 (25 seats)
Amount owed is $400.00
You earned 0 credits 
[othello, 30] => Statement for BigCo
  Othello: $400.00 (30 seats)
Amount owed is $400.00
You earned 0 credits 
[othello, 35] => Statement for BigCo
  Othello: $450.00 (35 seats)
Amount owed is $450.00
You earned 5 credits 
[othello, 40] => Statement for BigCo
  Othello: $500.00 (40 seats)
Amount owed is $500.00
You earned 10 credits 
[othello, 45] => Statement for BigCo
  Othello: $550.00 (45 seats)
Amount owed is $550.00
You earned 15 credits 
